drivers/hopwa_caper/app/models/hopwa_caper/enrollment.rb

```diff
@@ -0,0 +1,130 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HopwaCaper
+  class Enrollment < GrdaWarehouseBase
+    include SqlHelper
+    self.table_name = 'hopwa_caper_enrollments'
+
+    has_many :services, class_name: 'HopwaCaper::Service', foreign_key: [:hud_enrollment_id, :data_source_id, :report_instance_id], primary_key: [:hud_enrollment_id, :data_source_id, :report_instance_id]
+
+    def self.as_report_members
+      all.map do |record|
+        ::HudReports::UniverseMember.new(
+          universe_membership_type: sti_name,
+          universe_membership_id: record.id,
+        )
+      end
+    end
+
+    scope :overlapping_range, ->(start_date:, end_date:) {
+      where('entry_date <= :end_date AND (exit_date >= :start_date OR exit_date IS NULL)', start_date: start_date, end_date: end_date)
+    }
+
+    scope :latest_by_personal_id, -> {
+      select('DISTINCT ON (hopwa_caper_enrollments.data_source_id, hopwa_caper_enrollments.hud_personal_id) *').order(data_source_id: :asc, hud_personal_id: :asc, entry_date: :desc, id: :desc)
+    }
+
+    def self.head_of_household
+      where(relationship_to_hoh: 1)
+    end
+
+    INSURANCE_FIELDS = [
+      :Medicaid,
+      :Medicare,
+      :VAMedicalServices,
+      :HIVAIDSAssistance,
+      :SCHIP,
+      :RyanWhiteMedDent,
+    ].freeze
+    INCOME_SOURCE_FIELDS = [
+      :Earned,
+      :SocSecRetirement,
+      :SSI,
+      :SSDI,
+      :SNAP, :WIC, :TANF,
+      :PrivateDisability,
+      :VADisabilityService, :VADisabilityNonService,
+      :ChildSupport, :Alimony,
+      :WorkersComp,
+      :Unemployment,
+      :OtherIncomeSource
+    ].freeze
+    def self.from_hud_record(enrollment:, report:)
+      project = enrollment.project
+      hiv_disabilities = enrollment.disabilities.filter(&:hiv?).sort_by(&:id)
```

Comment:
Not sure if the intention is to order these newest first (or last) but sorting by ID won't guarantee that in any way, you probably want to sort by `InformationDate` (or `information_date`)

Reply:
thanks for pointing that out. I wasn't aiming for chronological order, just deterministic since by default it would be unordered